### PR TITLE
Merge completed Feature/specs 

### PR DESCRIPTION
**Pull request by sebGilR**
Method specifications were created for all enumerable methods, in a collaborative project.

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        expect(array_str.my_all? { |i| i =~ /./ }).to eq(true)
+      end
+
+      it 'returns false if not all elements in array fits the contition' do
+        expect(array_str.my_all? { |i| i =~ /^[a-zA-Z]*$/ }).to eq(false)
+      end
+    end
+
+    context 'there is no block given' do
+      it 'will return true if all the items are not nil nor false' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:57]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        expect(numbers.my_all?).to be_truthy
+      end
+
+      it 'will return true if all the elements are of a given class' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:65]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        expect(numbers.my_all?(Integer)).to eq(true)
+      end
+
+      it 'will return true if all items match regular expression' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:73]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        expect(array_str.my_any? { |i| i =~ /^[a-zA-Z]*$/ }).to eq(true)
+      end
+
+      it 'returns false if none of the elements fits the condition' do
+        expect(numbers.my_any? { |i| i =~ /^[a-zA-Z]*$/ }).to eq(false)
+      end
+    end
+
+    context 'there is no block given' do
+      it 'will return true if any the items is truthy' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:95]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        expect(numbers_nil.my_any?).to eql(true)
+      end
+
+      it 'will return true if any element is of a given class' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:103]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        expect(['s', 's', 3, 's'].my_any?(Integer)).to eq(true)
+      end
+
+      it 'will return true any item match regular expression' do
```
Comment (mosesogwo) [inline spec/main_spec.rb]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        expect(numbers.my_none? { |i| i =~ /^[a-zA-Z]*$/ }).to eq(true)
+      end
+
+      it 'returns false if any of the elements fit the condition' do
+        expect(array_str.my_none? { |i| i =~ /^[a-zA-Z]*$/ }).to eq(false)
+      end
+    end
+
+    context 'there is no block given' do
+      it 'will return true if none of the items is truthy' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:133]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        expect([nil, false].my_none?).to eql(true)
+      end
+
+      it 'will return true if any element is of a given class' do
```
Comment (mosesogwo) [inline spec/main_spec.rb]: Add test for the converse of this case.

File: spec/main_spec.rb [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        expect(%w[s s 3 s].my_none?(Numeric)).to eq(true)
+      end
+
+      it 'will return true if none if the items match regular expression' do
```
Comment (mosesogwo) [inline spec/main_spec.rb:150]: Add test for the converse of this case.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @sebGilR, @carlosprietobarron:clap:. Great job on your tests :+1:**

- Let's add some additional tests. See my comments on your code.

- Add instruction on how to run tests to your README.


_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by shshamim63** [comment]
## Status :100: Approved :100: 
___
Hi @sebGilR  and @carlosprietobarron :wave: ,
Awesome job :clap: . The project contains the instruction to run the test. And you have covered all the changes asked by the previous reviewer. So nothing is blocking me at this moment. This :+1: is from me. You can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Pull request merged**